lib/views/pay/pay_view.dart

```diff
@@ -1,114 +1,255 @@
+              enabledBorder: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(20),
+                borderSide: const BorderSide(
+                  color: Colors.white,
+                  width: 1.0,
+                ),
               ),
-              const Text(
-                'Enter milli-satoshi amount?',
-                style: TextStyle(fontSize: 17.0),
-              ), //Text//SizedBox
-            ],
-          ),
-          TextField(
-            keyboardType: TextInputType.number,
-            inputFormatters: <TextInputFormatter>[
-              FilteringTextInputFormatter.allow(RegExp(r'[0-9]')),
-            ],
-            onChanged: (amount) {
-              amountMsat = int.parse(amount);
-            },
-            enabled: amountMsat == null ? false : true,
-            decoration: const InputDecoration(
-              border: OutlineInputBorder(),
-              disabledBorder: OutlineInputBorder(),
-              labelText: 'Amount to send in millisatoshi',
-              hintText: 'Eg: 100 msats',
+              focusedBorder: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(20),
+                borderSide: const BorderSide(
+                  color: Color.fromARGB(255, 255, 121, 197),
+                  width: 1.0,
+                ),
+              ),
+              alignLabelWithHint: true,
+              contentPadding: const EdgeInsets.all(20),
+              labelText: 'Invoice Bolt11',
+              hintText: 'invoice',
             ),
           ),
           MainCircleButton(
-              icon: Icons.send_outlined,
-              label: "Pay",
-              onPress: () {
-                if (boltString == null || boltString!.isEmpty) {
-                  setState(() {
-                    error = "Error: Bolt11/12 Invoice required";
-                  });
-                } else {
-                  payInvoice(boltString!, amountMsat).then((value) => {
-                        setState(() {
-                          paymentResponse = value;
-                          error = null;
-                        }),
-                      });
+              icon: Icons.copy,
+              label: "Paste from clipboard",
+              width: 200,
+              onPress: () async {
+                ClipboardData? data = await Clipboard.getData('text/plain');
+                setState(() {
+                  _invoiceController.text = data!.text!;
+                });
+                boltString = data!.text;
+                AppDecodeInvoice invoice =
+                    await decodeInvoice(_invoiceController.text);
+                setState(() {
+                  display = invoice.invoice.amount.toString();
+                });
+                if (context.mounted) {
+                  _showBottomDialog(context: context, invoice: invoice);
                 }
               }),
-          error == null ? Text(state) : Text(error!)
```

Comment:
ok we are not catching the error anymore right? we should open a issue and make this working in the next iteration